Common/TableProducer/multiplicityTable.cxx

```diff
@@ -68,42 +95,85 @@ struct MultiplicityTableTaskIndexed {
       multZNC = zdc.energyCommonZNC();
     }
 
-    LOGF(debug, "multV0A=%5.0f multV0C=%5.0f multT0A=%5.0f multT0C=%5.0f multZNA=%6.0f multZNC=%6.0f multTracklets=%i multTPC=%i", multV0A, multV0C, multT0A, multT0C, multZNA, multZNC, multTracklets, multTPC);
-    mult(multV0A, multV0C, multT0A, multT0C, multZNA, multZNC, multTracklets, multTPC);
+    LOGF(debug, "multV0A=%5.0f multV0C=%5.0f multT0A=%5.0f multT0C=%5.0f multFDDA=%5.0f multFDDC=%5.0f multZNA=%6.0f multZNC=%6.0f multTracklets=%i multTPC=%i", multV0A, multV0C, multT0A, multT0C, multFDDA, multFDDC, multZNA, multZNC, multTracklets, multTPC);
+    mult(multV0A, multV0C, multT0A, multT0C, multFDDA, multFDDC, multZNA, multZNC, multTracklets, multTPC, multNContribs,
+         multZeqV0A, multZeqT0A, multZeqT0A, multZeqFDDA, multZeqFDDC, multZeqNContribs);
   }
   PROCESS_SWITCH(MultiplicityTableTaskIndexed, processRun2, "Produce Run 2 multiplicity tables", true);
 
-  void processRun3(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, soa::Join<aod::Tracks, aod::TracksExtra> const& tracksExtra, aod::BCs const& bcs, aod::Zdcs const& zdcs, aod::FV0As const& fv0as, aod::FT0s const& ft0s)
+  void processRun3(soa::Join<aod::Collisions, aod::EvSels>::iterator const& collision, soa::Join<aod::Tracks, aod::TracksExtra> const& tracksExtra, aod::BCs const& bcs, aod::Zdcs const& zdcs, aod::FV0As const& fv0as, aod::FT0s const& ft0s, aod::FDDs const& fdds)
   {
-    float multV0A = -1.f;
-    float multV0C = -1.f;
-    float multT0A = -1.f;
-    float multT0C = -1.f;
-    float multZNA = -1.f;
-    float multZNC = -1.f;
-    int multTracklets = -1;
+    float multFV0A = 0.f;
+    float multFV0C = 0.f;
+    float multFT0A = 0.f;
+    float multFT0C = 0.f;
+    float multFDDA = 0.f;
+    float multFDDC = 0.f;
+    float multZNA = 0.f;
+    float multZNC = 0.f;
+    int multTracklets = 0;
+
+    float multZeqFV0A = 0.f;
+    float multZeqFT0A = 0.f;
+    float multZeqFT0C = 0.f;
+    float multZeqFDDA = 0.f;
+    float multZeqFDDC = 0.f;
+    float multZeqNContribs = 0.f;
+
     auto tracksGrouped = tracksWithTPC->sliceByCached(aod::track::collisionId, collision.globalIndex());
+    auto pvContribsGrouped = pvContribTracks->sliceByCached(aod::track::collisionId, collision.globalIndex());
     int multTPC = tracksGrouped.size();
+    int multNContribs = tracksGrouped.size();
+
+    TList* lCalibObjects = ccdb->getForTimeStamp<TList>("Users/v/victor/Centrality/Calibration", 1635634560883); //temporary
+
+    TProfile* hVtxZFV0A = (TProfile*)lCalibObjects->FindObject("hVtxZFV0A");
+    TProfile* hVtxZFT0A = (TProfile*)lCalibObjects->FindObject("hVtxZFT0A");
+    TProfile* hVtxZFT0C = (TProfile*)lCalibObjects->FindObject("hVtxZFT0C");
+    TProfile* hVtxZFDDA = (TProfile*)lCalibObjects->FindObject("hVtxZFDDA");
+    TProfile* hVtxZFDDC = (TProfile*)lCalibObjects->FindObject("hVtxZFDDC");
+    TProfile* hVtxZNTracks = (TProfile*)lCalibObjects->FindObject("hVtxZNTracks");
```

Comment:
Do this on each collision?

Reply:
Thanks for taking a look, @jgrosseo! As I understand it, ccdb caching prevents a ccdb query (and this cannot be done at the Init level). it could also be done such that the TProfiles are pointers in the task struct, adding an "if ( == 0x0)" to assign them inside the process. Should I do that? My only question, then, is: will the timestamp range for the analysed data not potentially change to another range of validity?

Reply:
This is a side effect of having a TList. If the TProfile would be directly from CCDB then the update would happen on that pointer (and you would not need to do anything)...
I wonder how long the FindObject with string (x5) takes per collision...

Reply:
@ddobrigk @jgrosseo This part is still premature: temporary CCDB path and hardcoded timestamp
I would suggest the final status to be the same as for the centrality task where only when the run number has changed the CCDB is queried and the profiles stored.
Having different objects with the same time stamp will need to involve the meta-data and that will complicate things unnecessarily

Reply:
@ddobrigk @jgrosseo I meant this concrete section of the code it is not still in its final shape.
The most important thing, I guess, is not having too long this PR open, we risk the changes will need to be redone and it is a huge work.
I would suggest to inhibit this part of the code and get the PR approved with the changes in all involved sources and the new tables layout and then evolve this part towards its final shape